geopandas/tools/hilbert_curve.py

```diff
@@ -103,6 +103,8 @@ def _continuous_to_discrete(vals, val_range, n):
 
     """
     width = val_range[1] - val_range[0]
+    if width == 0:
```

Comment:
Does this need some kind of tolerance in comparing to zero, or is there only an issue if it's exactly zero. If width is almost zero, then 1/width is quite large, but n is also large so it might be fine (I'm not at all across the details of how hilbert curve is implemented)

Reply:
I think the warning from numpy will only happens if it's actually equal to zero (otherwise you don't get "division by zero" error and you don't introduce an Infinity, but just get a huge value; and that large value will get smaller by multiplying with `(vals - val_range[0])` and will otherwise get clipped anyway.

To be honest, this is also not the most important change, it's very much a corner case that it is fixing